reconcile/utils/terraform_client.py

```diff
@@ -531,34 +531,48 @@ def cleanup(self):
         for _, wd in self.working_dirs.items():
             shutil.rmtree(wd)
 
-    def _is_ignored_rds_modification(
+    def _get_pending_rds_modification(
```

Comment:
I know there is 0 test coverage for this method, but why leave it that way?  :) 
Since you are changing it drastically, could you please add a test for this method?

Reply:
@janboll, it was @geoberle who gave me a cautionary state, so to speak, when I suggested mocking AWS API response. Nonetheless, tests should be added, rightfully so.

I will refactor the code a bit to make it easier to test. Please, bear with me. :)

Reply:
@janboll, this turned out to be more involved to test, especially to do it properly - for instance, moto does not seem to include the **PendingModifiedValues** attribute as part of its RDS database mocked object.

This would require much refactoring, which would delay the value this change aims to bring. And we already have two RDS instances reporting different types of pending changes in our reconciliation log (which generates a lot of unwanted noise).

Additionally, @geoberle did a quick review and raised a question about why we even make an API call to AWS, given that Terraform does it too. We then parse pending information it provides - this is an excellent question. I will talk about this with @steveteahan, if he has a minute, to better understand some of the ideas behind the initial implementation. But this is also out of scope here.

Reply:
I've added tests that cover the _get_pending_rds_modification(). Aside of that, I've manually tested with in staging and verified using dry-run mode.